core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java

```diff
@@ -44,6 +41,42 @@ protected String cacheName() {
       return "DistributedExecutorFailoverTest";
    }
 
+   public void testBasicLocalDistributedCallable() throws Exception {
+       long taskTimeout = TimeUnit.SECONDS.toMillis(15);
+      EmbeddedCacheManager cacheManager1 = manager(0);
+      final EmbeddedCacheManager cacheManager2 = manager(1);
+      final EmbeddedCacheManager cacheManager3 = manager(2);
+      TestingUtil.killCacheManagers(cacheManager2);
+      TestingUtil.killCacheManagers(cacheManager3);
+      Cache<Object, Object> cache1 = cacheManager1.getCache();
+      DistributedExecutorService des = null;
+
+      try {
+         // initiate task from cache1 and execute on same node
+         des = new DefaultExecutorService(cache1);
+
+         SameNodeTaskFailoverPolicy sameNodeTaskFailoverPolicy = new SameNodeTaskFailoverPolicy(2);
+         DistributedTaskBuilder<Void> builder = des
+                 .createDistributedTaskBuilder(new TestCallable())
+                 .failoverPolicy(sameNodeTaskFailoverPolicy)
+                 .timeout(taskTimeout, TimeUnit.MILLISECONDS);
+
+         CompletableFuture<Void> future = des.submit(builder.build());
+         future.whenComplete((aVoid,throwable) -> {
+            Executors.newSingleThreadExecutor().submit((Callable<Void>)() -> {
+                future.get();
```

Comment:
This should be on the main thread, and it should have a timeout, e.g. `future.get(10, SECONDS)`.